Threads

mohamedfarith opened the pull request:
This pull request executes different ways of implementing threads
-->Makes API call and retrieve JSON data and display it in a recycler view.
-->Uses third-party library Retrofit to make API call.

mohamedfarith committed:
Different ways of threading in two activity app

mohamedfarith committed:
Merge branch 'master' into Threads

mohamedfarith committed:
-->Merge branch 'master' into Threads
-->This code is implemented for understanding different ways to executing background task
-->Implements threads by extending thread class and runnable interface

mohamedfarith committed:
-->This commit reolves the previous commit's build issue as it was pushed without checking for project build
-->Implemented different ways of threading like extending thread class,implementing runnable interface,making a network call
   using async task without using library.

mohamedfarith committed:
This commit implements making a network call using retrofit library

mohamedfarith committed:
Merge branch 'master' into Threads

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/MainActivity.java:
```diff
@@ -42,6 +48,10 @@ public void onClick(View view) {
                 Intent intentFragments = new Intent(MainActivity.this, FragmentActivity.class);
                 startActivity(intentFragments);
+                break;
+            case R.id.thread_button:
```
Can you call out different ways of running a background thread in android ?

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/Constants.java:
```diff
@@ -3,7 +3,11 @@
     public static final String KEY_NAME = "Name";
     public static final String LIST = "Name List";
+    public static final String THREAD = "ThreadActivityExtending";
+    public static final String WEATHER_URL = "https://andfun-weather.udacity.com/staticweather";
```
Why are there two urls ?

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginActivity.java:
```diff
@@ -15,78 +15,80 @@
 import static com.example.admin.learningandroid.Constants.SAVE_DATA;
 
-public class ModelLoginActivity extends AppCompatActivity implements View.OnClickListener {
-    public EditText etName;
```
This class should not have any changes ? Can you please back merge

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCall.java:
```diff
@@ -0,0 +1,15 @@
+import static com.example.admin.learningandroid.Constants.BASE_WEATHER_URL;
+
+public interface ApiCall {
+    String weatherURL = "https://andfun-weather.udacity.com/";
```
This is not used.

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java:
```diff
@@ -0,0 +1,142 @@
+                .addConverterFactory(GsonConverterFactory
+                        .create()).build();
+        ApiCall apiCall = dataRetreive.create(ApiCall.class);
+        Call<ListDataRetrieveModelClass> listDataRetrieveModelClassCall = apiCall.getWeatherData();
```
ListDataRetrieveModelClass is a  class its self explanatory please avoid appending class prefix

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java:
```diff
@@ -0,0 +1,142 @@
+                //Getting the list Array by creating an object of ListDataRetreiveModelClass
+
+
+                ;
```
?

nmrafiq87 replied:
Please remove unwanted space.

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java:
```diff
@@ -0,0 +1,142 @@
+
+
+                ;
+                for (int i = 0; i < listDataRetrieveModelClass.getList().size(); i++) {
```
why can't you use the same list to refresh data. Instead you are iterating through the contents and adding it in a different list and passing that to the recycler view.

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java:
```diff
@@ -0,0 +1,142 @@
+
+            @Override
+            public void onFailure(Call<ListDataRetrieveModelClass> call, Throwable t) {
+                String message = "no network connection";
```
The strings that are used for display should be kept in Resources

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/ApiCallUsingRetrofit.java:
```diff
@@ -0,0 +1,142 @@
+    public void onBindViewHolder(WeatherListViewHolder holder, int position) {
+
+        WeatherAPIModelClass modelClass = weatherList.get(position);
+        String message = "pressure: " + modelClass.getPressure();
```
you can set the string directly rather than creating so many string objects

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/MakeAPICallActivity.java:
```diff
@@ -0,0 +1,188 @@
+
+    public String getResponse() throws IOException {
+
+        URL weatherURL = new URL(WEATHER_URL);
```
Can there be on Entity for weather object and one adapter since the contents are the same

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/entity/RetrofitWeatherApiModelClass.java:
```diff
@@ -0,0 +1,39 @@
+
+public class RetrofitWeatherApiModelClass {
+    String pressure;
+    String humidity;
```
Please parse the entire response. Do not parse it selectively.

nmrafiq87 commented on app/src/main/java/com/example/admin/learningandroid/threads/MakeAPICallActivity.java:
```diff
@@ -0,0 +1,188 @@
+            RecyclerView weatherDataRecyclerView = findViewById(R.id.weather_api_data_recycler_view);
+            weatherDataRecyclerView.setLayoutManager
+                    (new LinearLayoutManager(MakeAPICallActivity.this));
+            weatherDataRecyclerView.setAdapter(new WeatherDataRecyclerViewAdapter(weatherDataList));
```
What will happen if the weatherdatalist is null here ?

mohamedfarith committed:
-->Replaced multiple adapters for weather data with a single adapter.
-->Replaced multiple model classes with a single model class.
-->Back merge Threads branch with master was checked.
-->Issue with null vlaue at ArrayList-weatherList resolved.
